lib/Updater.php

```diff
@@ -404,7 +411,7 @@ public function createBackup(): void {
 		 * @var string $path
 		 * @var \SplFileInfo $fileInfo
 		 */
-		foreach ($this->getRecursiveDirectoryIterator($currentDir) as $path => $fileInfo) {
+		foreach ($this->getRecursiveDirectoryIterator($currentDir, $excludedElements) as $path => $fileInfo) {
```

Comment:
Is it not dangerous that this excludes any directory named data even in subdirectories? How can we know that no Nextcloud application use a directory named data?

It’s not even clear to me whether this really excludes what’s in data. Because it applies the filter iterator around the recursive one, so it will recurse anyway and only filter the data dir itself but not children, no?